pyro/distributions/conjugate.py

```diff
@@ -85,3 +85,63 @@ def enumerate_support(self, expand=True):
         if expand:
             values = values.expand((-1,) + self._batch_shape)
         return values
+
+
+class GammaPoisson(TorchDistribution):
+    r"""
+    Compound distribution comprising of a gamma-poisson pair, also referred to as
+    a gamma-poisson mixture. The ``rate`` parameter for the
+    :class:`~pyro.distributions.Poisson` distribution is unknown and randomly
+    drawn from a :class:`~pyro.distributions.Gamma` distribution.
+
+    .. note:: This can be treated as an alternate parametrization of the
+        :class:`~pyro.distributions.NegativeBinomial` (``total_count``, ``probs``)
```

Comment:
I don't think `NegativeBinomial` is implemented yet. However I think it's worth mentioning the relation to `NegativeBinomial` (or implementing it as a simple wrapper) so that people can find it rather than needing to reimplement it.

Reply:
[Negative Binomial](https://pytorch.org/docs/stable/_modules/torch/distributions/negative_binomial.html) is implemented in `torch.distributions`! Do you think we should call that internally instead?

Reply:
Any idea why it doesn't appear in our docs? Even in dev http://docs.pyro.ai/en/dev/distributions.html

Re-implementing `GammaPoisson` is fine by me.

Reply:
@jpchen is just updating our rtd build so it should show up soon.

Reply:
Thanks for explaining.